worker/rhpv2.go

```diff
@@ -637,7 +638,21 @@ func (w *worker) fetchContractRoots(t *rhpv2.Transport, rev *rhpv2.ContractRevis
 func (w *worker) withTransportV2(ctx context.Context, hostKey types.PublicKey, hostIP string, fn func(*rhpv2.Transport) error) (err error) {
 	conn, err := dial(ctx, hostIP)
 	if err != nil {
-		return err
+		// if we can't resolve host IP, try last known IP as fallback
+		host, err := w.bus.Host(ctx, hostKey)
```

Comment:
@ChrisSchinnerl a thought I had here was to fetch the host from the bus at all times and essentially remove `hostIP` from our interfaces and request objects and such and use the host key to fetch the `api.Host` which then has all the information to properly fallback when dialing. Friction in many places comes from `api.ContractMetadata` which has both `hostIP` and `siamuxAddress`, deprecating those two fields would clean things up I feel?